statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/jsonmodule/JsonModule.java

```diff
@@ -49,10 +50,17 @@ public JsonModule(JsonNode moduleSpecNode, FormatVersion formatVersion, URL modu
 
   public void configure(Map<String, String> conf, Binder binder) {
     try {
-      ENTITIES.forEach(jsonEntity -> jsonEntity.bind(binder, moduleSpecNode, formatVersion));
+      ENTITIES.forEach(
+          jsonEntity ->
+              jsonEntity.bind(binder, getExtensionResolver(binder), moduleSpecNode, formatVersion));
     } catch (Throwable t) {
       throw new ModuleConfigurationException(
           format("Error while parsing module at %s", moduleUrl), t);
     }
   }
+
+  // TODO expose ExtensionResolver properly once we have more usages
+  private static ExtensionResolver getExtensionResolver(Binder binder) {
+    return (ExtensionResolver) binder;
```

Comment:
Notice this TODO.

I have considered add a `getExtensionResolver()` method on the binder interface of `StatefulFunctionModule.Binder`, but wasn't sure if it the right move given how few usages we have so far.